### PR TITLE
Allow PolygonSelector points to be removed

### PR DESCRIPTION
**Pull request by dstansby**
## PR Summary
This allows one to remove individual points of a `PolygonSelector` by right-clicking on them (mouse button 3).

## PR Checklist

- [x] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [x] New features are documented, with examples if plot related.
- [x] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [x] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [x] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [x] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

File: lib/matplotlib/widgets.py [hunk 1]
```diff
@@ -2794,6 +2800,34 @@ def __init__(self, ax, onselect, useblit=False,
...
+            self._ys.pop(-1)
+            # Close the polygon again by appending the new first vertex to the
+            # end
+            self._xs.append(self._xs[0])
+            self._ys.append(self._ys[0])
+        else:
+            self._xs.pop(i)
+            self._ys.pop(i)
+        if self._nverts <= 2:
+            # If only one point left, return to un-complete state to let user
```
Comment (QuLogic) [inline lib/matplotlib/widgets.py:2827]:
```suggestion
            # If only one point left, return to incomplete state to let user
```

File: lib/matplotlib/widgets.py [hunk 2]
```diff
@@ -2794,6 +2800,34 @@ def __init__(self, ax, onselect, useblit=False,
...
+            # start drawing again
+            self._polygon_completed = False
```
Comment (QuLogic) [inline lib/matplotlib/widgets.py:2829]: Should add a test for this.

**Comment by dstansby** [comment]
Thanks for the review. I've:
- Fixed typos
- Added a test for re-drawing the selector when only one point is left
- Added a note in the docstring that about the re-drawing behaviour

File: lib/matplotlib/tests/test_widgets.py [hunk 3]
```diff
@@ -500,3 +506,57 @@ def test_MultiCursor(horizOn, vertOn):
...
+    ax = get_ax()
+
+    def onselect(vertices):
+        pass
+
+    tool = widgets.PolygonSelector(ax, onselect)
+    for (etype, event_args) in event_sequence:
+        do_event(tool, etype, **event_args)
+    # After removing two verts, only one remains, and the
+    # selector should be automatically resete
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_widgets.py:561]:
```suggestion
    # selector should be automatically reset.
```

**Review by QuLogic — APPROVED** [review]

**Comment by dstansby** [comment]
@jklymak was this marked as a draft because it needed a rebase (now done), or is there something else?

**Comment by jklymak** [comment]
Just the rebase.  

When looking at old PRs it is easy to hit "draft:false" in the search terms.  It is not easy to put "not:label" for the 10 labels that indicate author action needed.

**Comment by dstansby** [comment]
👍, just checking

**Comment by jklymak** [comment]
... when you clear the "needs rebase" totally feel free to move back to "Ready for Review".  Its just a  way to keep things off the radar until the author has cleared the issue.  Our PR backlog is still too big for this to be super effective, but I go through once a week or so and try and sort another dozen or so.

**Review by jklymak — APPROVED** [review]
I didn't test this, but looks good, and @QuLogic seemed to do a thorough review...

**Pull request merged**